src/zenml/integrations/kserve/steps/kserve_pytorch_deployer.py

```diff
@@ -0,0 +1,315 @@
+
+import os
+import tempfile
+from typing import Optional, cast
+
+from model_archiver.model_packaging import package_model  # type: ignore
+from model_archiver.model_packaging_utils import (
+    ModelExportUtils,  # type: ignore
+)
+from pydantic import BaseModel
+
+from zenml.artifacts.model_artifact import ModelArtifact
+from zenml.environment import Environment
+from zenml.integrations.kserve.model_deployers.kserve_model_deployer import (
+    DEFAULT_KSERVE_DEPLOYMENT_START_STOP_TIMEOUT,
+    KServeModelDeployer,
+)
+from zenml.integrations.kserve.services.kserve_deployment import (
+    KServeDeploymentConfig,
+    KServeDeploymentService,
+)
+from zenml.io import fileio
+from zenml.logger import get_logger
+from zenml.steps import (
+    STEP_ENVIRONMENT_NAME,
+    BaseStepConfig,
+    StepEnvironment,
+    step,
+)
+from zenml.steps.step_context import StepContext
+from zenml.utils.source_utils import get_source_root_path, is_inside_repository
+
+logger = get_logger(__name__)
+
+
+class KServePytorchDeployerStepConfig(BaseStepConfig):
+    """KServe PyTorch model deployer step configuration.
+
+    Attributes:
+        service_config: KServe deployment service configuration.
+        model_class_file:     Path to Python file containing model architecture.
+        handler:        TorchServe's handler file to handle custom TorchServe inference logic.
+        extra_files:    Comma separated path to extra dependency files.
+        model_version: Model version.
+        requirements_file: Path to requirements file.
+        torch_config: TorchServe configuration file path.
+    """
+
+    service_config: KServeDeploymentConfig
+    model_class_file: str = ""
+    handler: str = ""
+    extra_files: Optional[str] = None
+    requirements_file: Optional[str] = None
+    model_version: Optional[str] = "1.0"
+    torch_config: Optional[str] = None
+    timeout: int = DEFAULT_KSERVE_DEPLOYMENT_START_STOP_TIMEOUT
+
+
+class TorchModelArchiver(BaseModel):
+    """Model Archiver for PyTorch models.
+
+    Attributes:
+        model_name: Model name.
+        model_version: Model version.
+        serialized_file: Serialized model file.
+        handler: TorchServe's handler file to handle custom TorchServe inference logic.
+        extra_files: Comma separated path to extra dependency files.
+        requirements_file: Path to requirements file.
+        export_path: Path to export model.
+        runtime: Runtime of the model.
+        force: Force export of the model.
+        archive_format: Archive format.
+    """
+
+    model_name: str
+    serialized_file: str
+    model_file: str
+    handler: str
+    export_path: str
+    extra_files: Optional[str] = None
+    version: Optional[str] = None
+    requirements_file: Optional[str] = None
+    runtime: Optional[str] = "python"
+    force: Optional[bool] = None
+    archive_format: Optional[str] = "default"
+
+
+@step(enable_cache=False)
+def kserve_pytorch_model_deployer_step(
```

Comment:
This step feels like a lot of copy-paste from the other. I see two ways to solve this:
- integrate this code into `kserve_model_deployer_step()` using `if predictor == "torch"` or similar at the relevant places
- Move all the duplicate code to some util functions, e.g. `.../kserve/steps/kserve_step_utils.py`